src/main/java/io/openliberty/tools/intellij/actions/LibertyGeneralAction.java

```diff
@@ -159,14 +159,15 @@ protected void notifyError(String errMsg) {
     }
 
     /**
-     * Returns the IntelliJ terminal widget for the corresponding Liberty module
+     * Returns the Terminal widget for the corresponding Liberty module
      *
-     * @param createWidget create terminal widget if it does not already exist
+     * @param createWidget create Terminal widget if it does not already exist
      * @return ShellTerminalWidget
      */
     protected ShellTerminalWidget getTerminalWidget(boolean createWidget) {
-        ShellTerminalWidget widget = LibertyProjectUtil.getTerminalWidget(project, projectName, createWidget);
-        if (widget == null) {
+        ShellTerminalWidget widget = LibertyProjectUtil.getTerminalWidget(project, libertyModule, createWidget);
+        libertyModule.setShellWidget(widget);
```

Comment:
I think `setShellWidget()` should be called at the time we create a new terminal in `LibertyProjectUtil.getTerminalWidget()` instead of every time we try to retrieve the ShellTerminalWidget.

Reply:
Good idea, I've adjusted the code to incorporate this change